src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -267,13 +269,21 @@ public JiraSite(URL url, URL alternativeUrl, StandardUsernamePasswordCredentials
               groupVisibility, roleVisibility, useHTTPAuth, DEFAULT_TIMEOUT, DEFAULT_READ_TIMEOUT, DEFAULT_THREAD_EXECUTOR_NUMBER);
     }
 
-    @DataBoundConstructor
+    // Deprecate the previous constructor but leave it in place for Java-level compatibility.
+    @Deprecated
     public JiraSite(URL url, URL alternativeUrl, String credentialsId, boolean supportsWikiStyleComment, boolean recordScmChanges, String userPattern,
                     boolean updateJiraIssueForAllStatus, String groupVisibility, String roleVisibility, boolean useHTTPAuth, int timeout, int readTimeout, int threadExecutorNumber){
         this(url, alternativeUrl, CredentialsHelper.lookupSystemCredentials(credentialsId, url), supportsWikiStyleComment, recordScmChanges, userPattern,
              updateJiraIssueForAllStatus, groupVisibility, roleVisibility, useHTTPAuth, timeout, readTimeout, threadExecutorNumber);
     }
 
+    @DataBoundConstructor
+    public JiraSite(String url){
+        URL mainURL = toURL(url);
+        if (mainURL == null) throw new AssertionError("URL cannot be null");
```

Comment:
your `toUrl` method would throw `AssetionError` itself, so no need to null check here

Reply:
No Alternative URL is not mandatory.

Reply:
yep, sorry )